spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/RemoveJsonAttributesResponseBodyGatewayFilterFactory.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.filter.factory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.rewrite.ModifyResponseBodyGatewayFilterFactory;
+import org.springframework.cloud.gateway.filter.factory.rewrite.RewriteFunction;
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.http.MediaType;
+
+/**
+ * @author Marta Medio
+ */
+public class RemoveJsonAttributesResponseBodyGatewayFilterFactory extends
+		AbstractGatewayFilterFactory<RemoveJsonAttributesResponseBodyGatewayFilterFactory.FieldListConfiguration> {
+
+	public RemoveJsonAttributesResponseBodyGatewayFilterFactory(
+			ModifyResponseBodyGatewayFilterFactory modifyResponseBodyGatewayFilterFactory) {
+		this.modifyResponseBodyGatewayFilterFactory = modifyResponseBodyGatewayFilterFactory;
+	}
+
+	@Override
+	public ShortcutType shortcutType() {
+		return ShortcutType.GATHER_LIST_TAIL_FLAG;
+	}
+
+	@Override
+	public List<String> shortcutFieldOrder() {
+		return Arrays.asList("fieldList", "deleteRecursively");
+	}
+
+	@Override
+	public FieldListConfiguration newConfig() {
+		return new FieldListConfiguration();
+	}
+
+	@Override
+	public Class<FieldListConfiguration> getConfigClass() {
+		return FieldListConfiguration.class;
+	}
+
+	@Override
+	public GatewayFilter apply(FieldListConfiguration config) {
+		ModifyResponseBodyGatewayFilterFactory.Config modifyResponseBodyConfig = new ModifyResponseBodyGatewayFilterFactory.Config();
+		modifyResponseBodyConfig.setInClass(String.class);
+		modifyResponseBodyConfig.setOutClass(String.class);
+
+		RewriteFunction<String, String> rewriteFunction = (exchange, body) -> {
+			if (MediaType.APPLICATION_JSON.isCompatibleWith(exchange.getResponse().getHeaders().getContentType())) {
+				try {
+					ObjectMapper mapper = new ObjectMapper();
```

Comment:
Probably don't want to create an `ObjectMapper` on every request

Reply:
See change: https://github.com/spring-cloud/spring-cloud-gateway/pull/2742/commits/09fc2791b12863a304ac34a806f9e37000266b47

Reply:
This also bad `ObjectMapper` should be a bean (constructor parameter). This will not work with JSR 310 Date Time because it is missing Java 8 modules configuration. It can also change resulted JSON for example from pretty formatted to oneline JSON or reverse. You cannot configure it because it is hardcoded in the filter.